tedana/reporting/static_figures.py

```diff
@@ -463,3 +464,69 @@ def pca_results(criteria, n_components, all_varex, io_generator):
     pca_variance_explained_name = os.path.join(io_generator.out_dir, "figures", plot_name)
     plt.savefig(pca_variance_explained_name)
     plt.close()
+
+
+def plot_t2star_and_s0(
+    *,
+    io_generator: io.OutputGenerator,
+    mask: np.ndarray,
+    png_cmap: str,
+) -> None:
+    """Create T2* and S0 maps and histograms.
+
+    Parameters
+    ----------
+    io_generator : :obj:`tedana.io.OutputGenerator`
+        The output generator for this workflow
+    mask : (S,) :obj:`numpy.ndarray`
+        Binary mask used to apply to the data.
+    png_cmap : :obj:`str`
+        Name of the colormap to use for the T2* and S0 maps.
+    """
+    t2star_img = io_generator.get_name("t2star img")
+    s0_img = io_generator.get_name("s0 img")
+    mask_img = io.new_nii_like(io_generator.reference_img, mask.astype(int))
+    assert os.path.isfile(t2star_img), f"File {t2star_img} does not exist"
+    assert os.path.isfile(s0_img), f"File {s0_img} does not exist"
+
+    # Plot T2* and S0 maps
+    t2star_plot = f"{io_generator.prefix}t2star_brain.svg"
+    plotting.plot_stat_map(
+        t2star_img,
+        bg_img=None,
+        display_mode="mosaic",
+        symmetric_cbar=False,
+        cmap=png_cmap,
+        output_file=os.path.join(io_generator.out_dir, "figures", t2star_plot),
+    )
+
+    s0_plot = f"{io_generator.prefix}s0_brain.svg"
+    plotting.plot_stat_map(
+        s0_img,
+        bg_img=None,
+        display_mode="mosaic",
+        symmetric_cbar=False,
+        cmap=png_cmap,
+        output_file=os.path.join(io_generator.out_dir, "figures", s0_plot),
+    )
+
+    # Plot histograms
+    sns.set_style("whitegrid")
+
+    t2star_data = masking.apply_mask(t2star_img, mask_img)
+    t2star_histogram = f"{io_generator.prefix}t2star_histogram.svg"
+
+    fig, ax = plt.subplots(figsize=(10, 6))
+    sns.histplot(data=t2star_data, bins=100, kde=True, ax=ax)
+    ax.set_title("T2*", fontsize=20)
+    ax.set_xlabel("Seconds")
+    fig.savefig(os.path.join(io_generator.out_dir, "figures", t2star_histogram))
+
+    s0_data = masking.apply_mask(s0_img, mask_img)
+    s0_histogram = f"{io_generator.prefix}s0_histogram.svg"
+
+    fig, ax = plt.subplots(figsize=(10, 6))
+    sns.histplot(data=s0_data, bins=100, kde=True, ax=ax)
+    ax.set_title("S0", fontsize=20)
+    ax.set_xlabel("Arbitrary Units")
+    fig.savefig(os.path.join(io_generator.out_dir, "figures", s0_histogram))
```

Comment:
For the 3 echo test data, I noticed the ranges were all flat and this is correctable. The following just saves the histograms before the maps so that I can calculate the 2nd and 98th percentile values on `t2star_data` and `s0_data` to and to use those values to scale the colormap for the brain maps. The histograms are also off and I tried to use `binrange` in `sns.histplot` but it wasn't saving correctly and I don't know why.

```suggestion
    # Plot histograms
    sns.set_style("whitegrid")

    t2star_data = masking.apply_mask(t2star_img, mask_img)
    t2star_percentiles_2_98_99 = np.percentile(t2star_data, [2, 98, 99])
    t2star_histogram = f"{io_generator.prefix}t2star_histogram.svg"

    fig, ax = plt.subplots(figsize=(10, 6))
    sns.histplot(
        data=t2star_data, bins=100, kde=True, ax=ax
    )
    ax.set_title("T2*", fontsize=20)
    ax.set_xlabel("Seconds")
    fig.savefig(os.path.join(io_generator.out_dir, "figures", t2star_histogram))

    s0_data = masking.apply_mask(s0_img, mask_img)
    s0_percentiles_2_98_99 = np.percentile(s0_data, [2, 98, 99])
    s0_histogram = f"{io_generator.prefix}s0_histogram.svg"

    fig, ax = plt.subplots(figsize=(10, 6))
    sns.histplot(data=s0_data, bins=100, kde=True, ax=ax)
    ax.set_title("S0", fontsize=20)
    ax.set_xlabel("Arbitrary Units")
    fig.savefig(os.path.join(io_generator.out_dir, "figures", s0_histogram))

    # Plot T2* and S0 maps
    t2star_plot = f"{io_generator.prefix}t2star_brain.svg"
    plotting.plot_stat_map(
        t2star_img,
        bg_img=None,
        display_mode="mosaic",
        symmetric_cbar=False,
        cmap=png_cmap,
        vmin=t2star_percentiles_2_98_99[0],
        vmax=t2star_percentiles_2_98_99[1],
        output_file=os.path.join(io_generator.out_dir, "figures", t2star_plot),
    )

    s0_plot = f"{io_generator.prefix}s0_brain.svg"
    plotting.plot_stat_map(
        s0_img,
        bg_img=None,
        display_mode="mosaic",
        symmetric_cbar=False,
        cmap=png_cmap,
        vmin=s0_percentiles_2_98_99[0],
        vmax=s0_percentiles_2_98_99[1],
        output_file=os.path.join(io_generator.out_dir, "figures", s0_plot),
    )
```

Reply:
I dropped seaborn, so there'll be a conflict with your suggestion. I can implement the general idea though.

What do you mean by the histograms being "off"?

Reply:
This is what the S0 histogram looks like for the three-echo tests. This is a weird truncated dataset, but it would be better to scale to the 98th or 99th percentile so that the x-axis bins are more like from 0-30000. This should be possible with the `binrange` parameter but it wasn't altering the saved plot. Maybe it will be easier to scale the range of the bins outside of seaborn.
![s0_histogram](https://github.com/ME-ICA/tedana/assets/7406227/b46451e3-14e5-490b-b17e-17fb65e0779f)

Reply:
I've added the 98th percentile as an upper limit to the figure, but TBH I think that the extreme values are informative.